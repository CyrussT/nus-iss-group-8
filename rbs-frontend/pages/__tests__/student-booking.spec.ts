import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import studentBooking from '../student-booking.vue';

describe('studentBooking', () => {
  it('renders correctly', () => {
    const wrapper = mount(studentBooking, {
      global: {
        stubs: {
          UButton: true,
          UIcon: true
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
