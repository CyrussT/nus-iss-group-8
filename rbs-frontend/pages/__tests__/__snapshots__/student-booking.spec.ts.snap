// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`studentBooking > renders correctly 1`] = `
"<div class="p-8">
  <h1 class="text-2xl font-bold mb-4">Booking Management</h1>
  <h3 class="text-2xl font-bold mb-4">My Bookings</h3><button aria-hidden="true" style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px;" type="button"></button>
  <div class="relative space-y-2 w-full">
    <div role="tablist" aria-orientation="horizontal" class="relative bg-gray-100 dark:bg-gray-800 rounded-lg p-1 w-full h-10 inline-grid items-center" style="grid-template-columns: repeat(3, minmax(0, 1fr));">
      <div class="absolute top-[4px] left-[4px] duration-200 ease-out focus:outline-none">
        <div class="w-full h-full bg-white dark:bg-gray-900 rounded-md shadow-sm"></div>
      </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-900 dark:text-white" id="headlessui-tabs-tab-v-0" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected">
        <u-icon-stub name="material-symbols:event-upcoming" class="w-4 h-4 flex-shrink-0 me-2"></u-icon-stub><span class="truncate text-primary-500 dark:text-primary-400">Upcoming</span>
      </button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-500 dark:text-gray-400" id="headlessui-tabs-tab-v-1" role="tab" aria-selected="false" tabindex="-1" data-headlessui-state="">
        <u-icon-stub name="material-symbols:calendar-clock" class="w-4 h-4 flex-shrink-0 me-2"></u-icon-stub><span class="truncate">Pending</span>
      </button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out h-8 px-3 text-sm font-medium rounded-md text-gray-500 dark:text-gray-400" id="headlessui-tabs-tab-v-2" role="tab" aria-selected="false" tabindex="-1" data-headlessui-state="">
        <u-icon-stub name="material-symbols:event-repeat" class="w-4 h-4 flex-shrink-0 me-2"></u-icon-stub><span class="truncate">Past</span>
      </button>
    </div>
    <div class="relative w-full">
      <div id="headlessui-tabs-panel-v-3" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="focus:outline-none">
        <div>
          <p class="font-bold mb-2">Upcoming Approved Bookings are shown here</p>
          <p>No upcoming bookings found.</p>
        </div>
      </div>
      <div id="headlessui-tabs-panel-v-4" role="tabpanel" tabindex="-1" hidden="" style="display: none;" data-headlessui-state="" class="focus:outline-none">
        <div>
          <p class="font-bold mb-2">Pending Bookings are shown here</p>
          <p>No pending bookings found.</p>
        </div>
      </div>
      <div id="headlessui-tabs-panel-v-5" role="tabpanel" tabindex="-1" hidden="" style="display: none;" data-headlessui-state="" class="focus:outline-none">
        <div>
          <p class="font-bold mb-2">Past Bookings</p>
          <p>No past bookings found.</p>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
